### PR TITLE
Cloning machine major fixes and changes

### LINKED ISSUE #4747: Prototype Cloner
Issue reported from  Round ID: 31094 (BeeStation Sage - Roleplay)

Reporting client version: 513.1542

The prototype cloning definitely doesn’t work anymore.
I blame the rework of the main cloning machin.

### LINKED ISSUE #5460: You can clone Xenomorphs and they'll pop out as humans.
image

They then got out as a human.

### LINKED ISSUE #5795: Experimental cloner will not clone
Issue reported from  Round ID: 34046 (BeeStation Sage - Roleplay)

Reporting client version: 514.1568

Experimental cloner doesn't clone anymore.

Found it in a random maint room, relocated it to medbay lobby and rebuilt with better parts. Built scanner -> console -> cloner. Cloner console recognizes the scanner and the pod. When attempting to clone it initiates scan, but does not complete scan and start cloning
Capture
.

### LINKED ISSUE #5997: Cloning only shows ''Body-Only''
Don't know when exactly this issue started to happen but at the current moment, when you select someone to clone them the only option that shows is ''Body only'', from what i've seen on **multiple** rounds this had lead to the following scenarios:

- Most doctors mains and new were equally confused about why only that option was showing up
- Brains sometimes didn't transfer the soul correctly to the new body, since well.. it does create a ''body only''.
- A lot of unespecified reasons for the body not being suited to be clonned (since empty clone works differently than regular iirc)

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Cloning machine fixes

## Why It's Good For The Game
- Fixes #5997
- FIxes #5795
- Fixes #5460
- Fixes #4747

Who made this shit

## Changelog
:cl:
refactor: Change to the cloning console and machine codes for error returns
fix: Cloning console TG window is now automatically updated when you trigger something.
tweak: Cloning console now tells which error you have failed to clone at.
tweak: Cloning console now properly lets you know that you can't clone a presaved person. It has always been so, but it has not warned you.
fix: Cloning record id is now reliably saved based on your name and mind, which fixes the record conflicts.
fix: Cloning console displays and accepts data in cloning diskettes properly, but remember that the clonable data becomes outdated once you clone them.
fix: Cloning console now can track the presaved person's health status.
tweak: Players can't turn on the autoclone manually anymore. admin still can turn it on by variable edit. (Note that Autoclone has not ever been working, ever.)
tweak: Body only clone will not transfer your quirks anymore and transferring your brain into that body will make you lose them. This is likely a bug due to the structure of quirks. Full scan clone still keeps them, so the brain transplant trick will not work.
fix: Experimental cloner will properly work now and they can save data records now.
add: Experimental clone ghost role added. It will find candidates to pick rather than just allowing all ghosts to come in. (*a clone being catatonic means no ghost wanted the role.)
fix: You're not allowed to scan xenomorphs anymore - they'll not be cloned as a human. Don't harass them.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
damn, TG check failed

**Comment by KubeRoot** [comment]
Use the exact syntax `Fixes #issue number` in your PR. If you want more details to be visible on the issue referenced, you can put it in a list, for example:

Fixes #5997

- Fixes #5997

**Comment by EvilDragonfiend** [comment]
I got approval from Archanial to remove the autoclone button. the feature is still in the code and admins can turn it on still by variable edit.

File: code/game/machinery/computer/cloning.dm
Comment (EvilDragonfiend) [inline code/game/machinery/computer/cloning.dm:592]: since id is more complex, we don't need to compare more things to check.

File: tgui/packages/tgui/interfaces/CloningConsole.js
Comment (EvilDragonfiend) [inline tgui/packages/tgui/interfaces/CloningConsole.js:38]: removal approved by Archanial

**Comment by EvilDragonfiend** [comment]
found a possible bug again

**Comment by EvilDragonfiend** [comment]
the possible bug is not from the cloner.

**Comment by yurara** [comment]
(This is EvilDragonfiend and I had to log in as this for a reason... I forgot to verify my account in my phone.)
Since this is likely a major fix for the unfunctional clone code, I'd like to ask the maintainers to accept this in a generous consideration.
The whole code actually should be rewritten, but it would need a lot of time and maintainers should review the rewritten clone code with a lot of their time too, so approving the fixing PR will be better for now.

File: code/game/machinery/cloning.dm
Comment (Archanial) [inline code/game/machinery/cloning.dm]: candidates is not lazy

Comment (EvilDragonfiend) [reply code/game/machinery/cloning.dm]: I just copy-pasted it from others. all other codes do the same thing. I am not sure if I should change that too since I know there's a PR that removed needless LAZYLENs.

Comment (Archanial) [reply code/game/machinery/cloning.dm]: length(candidates)

Comment (EvilDragonfiend) [reply code/game/machinery/cloning.dm]: I updated it, but not all LAZYLEN(candidates) at other codes since I don't want to test possible bugs as it is not a PR for candidates LAZYLEN fixes.

**Comment by EvilDragonfiend** [comment]
this PR is now ready for testmerge.
personally, I am not sure if I should change `body-only` term into `empty` as it was an original term used before. just keep `body-only`?

File: code/game/machinery/computer/cloning.dm
Comment (Archanial) [inline code/game/machinery/computer/cloning.dm]:
```suggestion
					return TRUE
				if(CLONING_SUCCESS_EXPERIMENTAL)
					temp = "Notice: [C.fields["name"]] => Experimental cloning cycle in progress..."
					playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
					return TRUE
```

File: code/game/machinery/computer/cloning.dm
Comment (Archanial) [inline code/game/machinery/computer/cloning.dm:601]: undocumented change

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/cloning.dm
Comment (EvilDragonfiend) [reply code/game/machinery/computer/cloning.dm:601]: - fix: Cloning record id is now reliably saved based on your name and mind, which fixes the record conflicts.

Do you want me to specify this more?

File: code/game/machinery/computer/cloning.dm
Comment (EvilDragonfiend) [reply code/game/machinery/computer/cloning.dm]: done. thanks.

**Review by Archanial — APPROVED** [review]

**Pull request merged**
